### PR TITLE
Immutable areas

### PR DESCRIPTION
**Pull request by wossnameGitHub**
Areas are now immutable (once again).

Changes:
- Area.con_left removed and replaced by dict 'connections_left'
- Area.exits is no longer changable. If we want to change it anyway we make a new Area object instead

### CODE REVIEW

File: Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py [hunk 1]
```diff
@@ -344,6 +355,7 @@ def get_random_redirection(original: Transition, all_redirections: Iterable[Tran
 
 def set_transitions_map():  # TODO: Break up in smaller functions
```
Comment (Avasam) [inline Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py:356]:
```suggestion
def set_transitions_map():  # noqa: PLR0915 # TODO: Break up in smaller functions
```

File: Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py [hunk 2]
```diff
@@ -67,6 +67,8 @@ class Choice(IntEnum):
 }
 ```"""
 
+connections_left: dict[int, int] = {}
```
Comment (Avasam) [inline Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py:70]: This is exclusively used for entrance rando generation to temporarily store a state right? If so let's make it private just so it doesn't show up for other modules. (feel free to reword the docstring).

```suggestion
__connections_left: dict[int, int] = {}
"""Used for tracking transitions to randomize during the randomization process."""
```

**Review by Avasam — CHANGES_REQUESTED** [review]
Should be a quick update and good to go!

As for the pyright issue, I'm fairly certain that has to be a false positive in 1.1.368 that just got published 18h ago. If you rebase/merge main I fixed it by locking the version in https://github.com/Avasam/ptle-tools/pull/55

File: Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py (hunk 1 above)
Comment (wossnameGitHub) [reply Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py:356]: Added.

File: Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py (hunk 2 above)
Comment (wossnameGitHub) [reply Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py:70]: Sure why not.

**Review by Avasam — APPROVED** [review]
Thanks!

**Pull request merged**